From 93e79fe76a20659e220cd7cfe1f8f8c90b650a07 Mon Sep 17 00:00:00 2001
From: carl <carl@cloudera.com>
Date: Mon, 8 Aug 2011 21:19:53 -0700
Subject: [PATCH 51/51] CLOUDERA-BUILD. Set mapred.max.split.size in hive-default.xml

Reason: HIVE-2037 neglected to include several property settings in hive-default.xml
Author: Carl Steinbach
Ref: CDH-3443, HIVE-2362
---
 conf/hive-default.xml |   25 +++++++++++++++++++++++++
 1 files changed, 25 insertions(+), 0 deletions(-)

diff --git a/conf/hive-default.xml b/conf/hive-default.xml
index 6efc095..b9600f3 100644
--- a/conf/hive-default.xml
+++ b/conf/hive-default.xml
@@ -37,6 +37,31 @@
 </property>
 
 <property>
+  <name>mapred.max.split.size</name>
+  <value>256000000</value>
+  <description>Largest valid size in bytes for a file split.
+  This should be set to a small multiple of the block size in bytes</description>
+</property>
+
+<property>
+  <name>mapred.min.split.size</name>
+  <value>1</value>
+  <description>Smallest valid size in bytes for a file split</description>
+</property>
+
+<property>
+  <name>mapred.min.split.size.per.rack</name>
+  <value>1</value>
+  <description>The minimum number of bytes of data required to create a rack-local partition</description>
+</property>
+
+<property>
+  <name>mapred.min.split.size.per.node</name>
+  <value>1</value>
+  <description>The minimum number of bytes of data required to create a node-local partition</description>
+</property>
+
+<property>
   <name>hive.exec.reducers.bytes.per.reducer</name>
   <value>1000000000</value>
   <description>size per reducer.The default is 1G, i.e if the input size is 10G, it will use 10 reducers.</description>
-- 
1.7.0.4

